ner_v2/detectors/base_detector.py

```diff
@@ -0,0 +1,180 @@
+import abc
+from language_utilities.constant import ENGLISH_LANG
+from constants.ner_constant import (FROM_STRUCTURE_VALUE_VERIFIED, FROM_STRUCTURE_VALUE_NOT_VERIFIED, FROM_MESSAGE,
+                                    FROM_FALLBACK_VALUE, ORIGINAL_TEXT, ENTITY_VALUE, DETECTION_METHOD,
+                                    DETECTION_LANGUAGE, ENTITY_VALUE_DICT_KEY)
+
+
+class BaseDetector(object):
```

Comment:
Not a single change required in this class except following when compared to the base detector in ner_v1
1) Rename source_language_script to 'language'
2) Rename target_language_script to 'processing_language'
Whatever has been changed in this file has two negative impact.
1) NotImplementedError is not handled centrally for all detectors. Technically date/date_detection.py should check folders and detectors, etc and add things to self.supported languages. It should generate logs of what is supported and raise warning in case of discrepancy where data is added and language_detector is missing, etc.
2) Translation is disabled. We need to keep it for emergency utility.

Reply:
done